src/AccountManager.ts

```diff
@@ -0,0 +1,128 @@
+import sigUtil, {
+  SignTypedDataVersion,
+  TypedMessage,
+} from '@metamask/eth-sig-util';
+import type { EnvelopingTypes } from '@rsksmart/rif-relay-contracts/dist/typechain-types/contracts/RelayHub';
+import { providers, Wallet } from 'ethers';
+import { getAddress } from 'ethers/lib/utils';
+import {
+  deployRequestType,
+  EnvelopingMessageTypes,
+  getEnvelopingRequestDataV4Field,
+  relayRequestType,
+} from './typedRequestData.utils';
+
+export type RelayOrDeployRequest =
+  | EnvelopingTypes.RelayRequestStruct
+  | EnvelopingTypes.DeployRequestStruct;
+
+export default class AccountManager {
+  private _provider: providers.Provider;
+
+  private _accounts: Wallet[] = [];
+
+  chainId: number;
+
+  constructor(provider: providers.Provider, chainId: number) {
+    this._provider = provider;
+    this.chainId = chainId;
+  }
+
+  getAccounts(): string[] {
+    return this._accounts.map((it) => it.address);
+  }
+
+  addAccount(account: Wallet): void {
+    const wallet = new Wallet(account.privateKey, this._provider);
+    if (wallet.address !== account.address) {
+      throw new Error('invalid keypair');
+    }
+    this._accounts.push(wallet);
+  }
+
+  isDeployRequest({ request }: RelayOrDeployRequest): boolean {
+    return 'recoverer' in request;
```

Comment:
I would use the index instead of the recoverer.

Reply:
1. explain
2. the original implementation uses recoverer

Reply:
The recoverer function is not implemented yet and it could be optional since we are using always the zero address, but the index cannot be optional at all since its necessary.